### PR TITLE
Ttv mb 1141 approving shipment creates rdd for conus moves

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description

Add ability to create a required delivery date that is populated in an MTO Shipment when we are provided with a scheduled pickup date.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Migrating the db:
```sh
make db_test_migrate
```
(for dev db):
```sh
make db_dev_migrate
```

Running the specific test:

```sh
go test ./pkg/services/mto_shipment -testify.m TestMTOShipmentUpdater
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/tree/master/docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/blob/master/docs/database/migrate-the-database.md#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1141) for this change

### CODE REVIEW

**Comment by Ryan-Koch** [comment]
> The test works, but when I approve a move from the TOO application, a required delivery date is not being generated. Am I missing a step to test this properly?

Ah I see what's happening here. On the `ghcapi` side there's a status update service object that also needs to be aware of this activity. I'll take a look at that and push something.

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 1]
```diff
@@ -326,6 +336,32 @@ func constructMTOServiceItemModels(shipmentID uuid.UUID, mtoID uuid.UUID, reServ
...
+	if err != nil {
+		return nil, err
+	}
+	// Query the ghc_domestic_transit_times table for the max transit time
+	var ghcDomesticTransitTime models.GHCDomesticTransitTime
+	err = db.Where("distance_miles_lower <= ? "+
+		"AND distance_miles_upper >= ? "+
+		"AND weight_lbs_lower <= ? "+
+		"AND weight_lbs_upper >= ?",
+		distance, distance, weight, weight).First(&ghcDomesticTransitTime)
```
Comment (donaldthai) [inline pkg/services/mto_shipment/mto_shipment_updater.go]: There's one caveat I'd like to mention about `weight_lbs_upper`. The column value may contain a `0`, which stands for no bounds.

File: pkg/services/mto_shipment/mto_shipment_updater.go [hunk 2]
```diff
@@ -326,6 +336,32 @@ func constructMTOServiceItemModels(shipmentID uuid.UUID, mtoID uuid.UUID, reServ
...
+
+	if err != nil {
+		return nil, err
+	}
+	// Add the max transit time to the pickup date to get the new required delivery date
+	requiredDeliveryDate := pickupDate.AddDate(0, 0, ghcDomesticTransitTime.MaxDaysTransitTime)
```
Comment (donaldthai) [inline pkg/services/mto_shipment/mto_shipment_updater.go]: We also have to add additional days if the address is in certain states and cities:
```
NOTE: For shipments to and from Alaska, transit times are constructed as follows:
--
a. For shipments to and from Anchorage, Fairbanks, and all other Alaska points (except Adak, Kodiak, Juneau, Ketchikan, Sitka and other cities located in the vicinity of these cities), ADD an additional 10 days for time in port to the applicable days shown above.
b. For shipments to and from Adak, Kodiak, Juneau, Ketchikan, Sitka and other cities located in the vicinity of these cities ADD an additional 20 days for time in port to the applicable days shown above.
```

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 1 above)
Comment (donaldthai) [inline pkg/services/mto_shipment/mto_shipment_updater.go]: There's one caveat I'd like to mention about weight_lbs_upper. The column value may contain a 0, which stands for no bounds so we would have to account for that.

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 2 above)
Comment (Ryan-Koch) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Ah I see. Thanks for pointing that out, I shall take a crack at it.

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 1 above)
Comment (Ryan-Koch) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Good point, I'll try and account for that case.

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 2 above)
Comment (Ryan-Koch) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Hm something I see as potentially difficult here is the 'vicinity of these cities' portion. We should try and get more detail there. The cities mentioned though shouldn't be a super heavy lift.

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 1 above)
Comment (Ryan-Koch) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Got this added in

File: pkg/services/mto_shipment/mto_shipment_updater.go (hunk 2 above)
Comment (Ryan-Koch) [reply pkg/services/mto_shipment/mto_shipment_updater.go]: Got it covered.

**Pull request merged**